jbmc/src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -1031,6 +1059,15 @@ static void notify_static_method_calls(
           expr_dynamic_cast<symbol_exprt>(fn_call->function());
         needed_lazy_methods->add_needed_method(fn_sym.get_identifier());
       }
+      else if(
```

Comment:
Separate commit sounds good, they're added because we now make side-effect-function-call expressions to avoid a pointless temporary (the expression is equivalent to a `code_function_callt` with a symbol on the LHS followed by immediately returning that symbol)